core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/AppCDSRequestedBuildItem.java

```diff
@@ -13,13 +13,23 @@ public final class AppCDSRequestedBuildItem extends SimpleBuildItem {
     /**
      * Directory where various files needed for AppCDS generation will reside
      */
-    private final Path appCDSDir;
+    private final Path dir;
+    private final JvmStartupOptimizerArchiveType type;
 
-    public AppCDSRequestedBuildItem(Path appCDSDir) {
```

Comment:
The name of the build item is not ideal if we want to make it something generic.